coalib/coala.py

```diff
@@ -59,11 +60,35 @@ def main(debug=False):
 
         if args.show_bears:
             from coalib.settings.ConfigurationGathering import (
-                get_filtered_bears)
-
-            local_bears, global_bears = get_filtered_bears(
-                args.filter_by_language, log_printer)
-
+                get_all_bears)
+            filtered_bears = get_all_bears(log_printer)
+            if args.filter_by_language:
+                # filter_by_language option should be deprecated soon.
+                # Since it is already covered by filter_by option.
+                log_printer.warn(
+                    "'--filter-by-language ...' option is deprecated. "
+                    "Use '--filter-by language ...' option instead")
+                args.filter_by = args.filter_by or []
```

Comment:
If args.filter_by is None, then it will be empty list after this statement.

This is to cover the following case:
`coala -B --filter-by-language java`

In this case, we first build `args.filter_by` from `args.filter_by_language`. And then process it, to make it simple. And it will also make it easier to remove `if args.filter_by_language:` in future as it will be deprecated after adding this generic filter processor.